distributed/worker.py

```diff
@@ -760,3 +763,41 @@ def apply_function(function, args, kwargs):
     msg['compute_stop'] = end
     msg['thread'] = current_thread().ident
     return msg
+
+
+def convert_args_to_str(args, max_len=1000):
+    """ Convert args to a string, allowing for some arguments to raise
+    exceptions during conversion and ignoring them.
+    """
+    sargs = "("
+    for arg in args:
+        try:
+            sarg = repr(arg)
+        except:
+            sarg = "< could not convert arg to str >"
+        sargs += sarg + ", "
```

Comment:
It is slightly more efficient if you make a list of strings and then create the result in one go with `join` as in `'(' + ', '.join(strings) + ')'` instead.  Otherwise we create many copies.  This isn't a big deal though (number of args is usually small).